CRM/Core/BAO/Email.php

```diff
@@ -339,6 +339,108 @@ public static function getFromEmail() {
+   * @throws \CiviCRM_API3_Exception
+   */
+  public static function getToEmail($contactID) {
+    $emails = [];
+
+    $contactEntity = civicrm_api3('Contact', 'getsingle', [
+      'sequential' => 1,
+      'id' => $contactID,
+      'api.Email.get' => ['api.LocationType.getsingle' => []],
+    ]);
+
+    $contactName = $contactEntity['display_name'];
+    //Get mail entities from this contact
+    $mailEntities = $contactEntity['api.Email.get'];
+    foreach ($mailEntities['values'] as $mailEntity) {
+      $email = $mailEntity['email'];
+      //Get location type entity associated
+      $locationTypeEntity = $mailEntity['api.LocationType.getsingle'];
+      $locationName = $locationTypeEntity['display_name'];
+
+      $toEmail = "$contactName <$email>";
+      $toEmailHtml = $locationName;
+
+      if (!empty($mailEntity['is_primary'])) {
+        $toEmailHtml .= ' ' . ts('(preferred) ');
+      }
+
+      $toEmailHtml .= htmlspecialchars($toEmail);
+
+      //Add contact email to mails array
+      $emails[$toEmail] = $toEmailHtml;
+    }
+    //Add contact email to mails array
+    $emails[$toEmail] = $toEmailHtml;
+
+    return $emails;
+  }
+
+  /**
+   * Get Email from multiple contacts to send contributions
+   *
+   * @param array $contactIDs
+   *
+   * @return array
+   * @throws \CiviCRM_API3_Exception
+   */
+  public static function getMultiEmails($contactIDs) {
+    $locations = [];
+    $contactsData = [];
+
+    $contactsEntity = civicrm_api3('Contact', 'get', [
+      'sequential' => 1,
+      'id' => ['IN' => $contactIDs],
+      'api.Email.get' => ['api.LocationType.getsingle' => []],
+    ]);
+
+    foreach ($contactsEntity['values'] as $contactEntity) {
+      //Get all available locations
+      $mailEntities = $contactEntity['api.Email.get'];
+      $contactID = $contactEntity['contact_id'];
+      $contactName = $contactEntity['display_name'];
+      foreach ($mailEntities['values'] as $mailEntity) {
+        $email = $mailEntity['email'];
+        $emailID = $mailEntity['id'];
+        //Get location type entity associated
+        $locationTypeEntity = $mailEntity['api.LocationType.getsingle'];
+        $locationName = $locationTypeEntity['display_name'];
+        $locations[] = $locationName;
+        $contactsData[$contactID][$emailID]['email'] = $email;
+        $contactsData[$contactID][$emailID]['is_preferred'] = $mailEntity['is_primary'];
+        $contactsData[$contactID][$emailID]['location_name'] = $locationName;
+        $contactsData[$contactID][$emailID]['contact_name'] = $contactName;
+      }
+    }
+
+    $locations = array_unique($locations);
+    $toEmails = [];
+
+    //Get mails for each location
+    foreach ($locations as $location) {
+      $emails = [];
+      foreach ($contactsData as $contactData){
+        $emails = array_merge(
+          static::getLocationOrPreferredMail($location, $contactData),
+          $emails
+        );
+      }
+      $emails = implode(", ", $emails);
```

Comment:
Hrm. I think this is not quite right. This appears to be setting up the email to go to both emails (ie. 'email1@example.com, email2@example.com'), rather than 2 individual emails to each of the recipients.

Reply:
This function (if it's even necessary) should just return the list of location, and not concrete email addresses